sql/core/src/main/scala/org/apache/spark/sql/execution/command/AlterTableCommandParser.scala

```diff
@@ -0,0 +1,410 @@
+            (Some(bucketSpec), false, false)
+          case Token("TOK_NOT_CLUSTERED", Nil) =>
+            (None, true, false)
+          case Token("TOK_NOT_SORTED", Nil) =>
+            (None, false, true)
+        }
+        AlterTableStoreProperties(
+          tableIdent,
+          buckets,
+          noClustered,
+          noSorted)(node.source)
+
+      case Token("TOK_ALTERTABLE_BUCKETS", Token(bucketNum, Nil) :: Nil) :: _ =>
+        val num = bucketNum.toInt
+        val buckets = Some(BucketSpec(num, Nil, Nil, Nil))
+        AlterTableStoreProperties(
+          tableIdent,
+          buckets,
+          noClustered = false,
+          noSorted = false)(node.source)
+
+      case Token("TOK_ALTERTABLE_SKEWED", Nil) :: _ =>
+        // ALTER TABLE table_name NOT SKEWED
+        AlterTableSkewed(
+          tableIdent,
+          Nil,
+          Nil,
+          storedAsDirs = false,
+          notSkewed = true,
+          notStoredAsDirs = false)(node.source)
+
+      case Token("TOK_ALTERTABLE_SKEWED", Token("TOK_STOREDASDIRS", Nil) :: Nil) =>
+        // ALTER TABLE table_name NOT STORED AS DIRECTORIES
+        AlterTableSkewed(
+          tableIdent,
+          Nil,
+          Nil,
+          storedAsDirs = false,
+          notSkewed = false,
+          notStoredAsDirs = true)(node.source)
+
+      case (tableSkewed @ Token("TOK_ALTERTABLE_SKEWED", _)) :: _ =>
+        val skewedArgs = tableSkewed match {
+          case Token("TOK_ALTERTABLE_SKEWED", args :: Nil) =>
+            args match {
+              case Token("TOK_TABLESKEWED", skewedCols :: skewedValues :: stored) =>
+                val cols = skewedCols.children.map(n => cleanAndUnquoteString(n.text))
+                val values = skewedValues match {
+                  case Token("TOK_TABCOLVALUE", colVal) =>
+                    Seq(colVal.map(n => cleanAndUnquoteString(n.text)))
+                  case Token("TOK_TABCOLVALUE_PAIR", pairs) =>
+                    pairs.map {
+                      case Token("TOK_TABCOLVALUES", colVals :: Nil) =>
+                        colVals match {
+                          case Token("TOK_TABCOLVALUE", vals) =>
+                            vals.map(n => cleanAndUnquoteString(n.text))
+                        }
+                    }
+                }
+
+                val storedAsDirs = stored match {
+                  case Token("TOK_STOREDASDIRS", Nil) :: Nil => true
+                  case _ => false
+                }
+
+                (cols, values, storedAsDirs)
+            }
+        }
+        val (cols, values, storedAsDirs) = skewedArgs
+        AlterTableSkewed(
+          tableIdent,
+          cols,
+          values,
+          storedAsDirs,
+          notSkewed = false,
+          notStoredAsDirs = false)(node.source)
+
+      case Token("TOK_ALTERTABLE_SKEWED_LOCATION",
+      Token("TOK_SKEWED_LOCATIONS",
+      Token("TOK_SKEWED_LOCATION_LIST", locationMaps) :: Nil) :: Nil) :: _ =>
```

Comment:
Isn't the wildcard used in the previous match preventing this from ever happening? Why not swap them?